includes/class-wc-stripe-payment-tokens.php

```diff
@@ -98,37 +98,21 @@ public function woocommerce_get_customer_payment_tokens( $tokens, $customer_id,
 				$stored_tokens[] = $token->get_token();
 			}
 
-			if ( 'stripe' === $gateway_id ) {
-				$stripe_customer = new WC_Stripe_Customer( $customer_id );
-				$stripe_sources  = $stripe_customer->get_sources();
+			$stripe_customer = new WC_Stripe_Customer( $customer_id );
 
-				foreach ( $stripe_sources as $source ) {
-					if ( isset( $source->type ) && 'card' === $source->type ) {
-						if ( ! in_array( $source->id, $stored_tokens ) ) {
-							$token = new WC_Payment_Token_CC();
-							$token->set_token( $source->id );
-							$token->set_gateway_id( 'stripe' );
-
-							if ( 'source' === $source->object && 'card' === $source->type ) {
-								$token->set_card_type( strtolower( $source->card->brand ) );
-								$token->set_last4( $source->card->last4 );
-								$token->set_expiry_month( $source->card->exp_month );
-								$token->set_expiry_year( $source->card->exp_year );
-							}
+			if ( 'stripe' === $gateway_id ) {
+				$payment_methods  = $stripe_customer->get_payment_methods( 'card' );
 
-							$token->set_user_id( $customer_id );
-							$token->save();
-							$tokens[ $token->get_id() ] = $token;
-						}
-					} else {
-						if ( ! in_array( $source->id, $stored_tokens ) && 'card' === $source->object ) {
+				foreach ( $payment_methods as $payment_method ) {
+					if ( isset( $payment_method->type ) && 'card' === $payment_method->type ) {
```

Comment:
Since we're querying only `cart` or `sepa_debit` now with `$stripe_customer->get_payment_methods( 'card' );`, maybe this check `isset( $payment_method->type ) && 'card' === $payment_method->type` is not necessary, but it might as well be a safety check there, so feel free to ignore.

Reply:
You are right, the check is not necessary